apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -2292,20 +2291,12 @@ private static TypeName selectJavaFieldType(
       return ClassName.get(LocalDate.class);
     } else if (type == RifColumnType.TIMESTAMP) {
       return ClassName.get(Instant.class);
-    }
-    // handle an inherited hack from the Excel spreadsheet in which a row entry
-    // was defined as a NUM and had an associated scale; for example (12,2) denotes
-    // a numeric data types of up to 12 digits, with two digits of scale (i.e., 55.45).
-    else if (type == RifColumnType.NUM && columnScale.orElse(Integer.MAX_VALUE) > 0) {
-      return ClassName.get(BigDecimal.class);
-    }
-    // some entries in Excel spreadsheet defined as NUM with a zero scale that are
-    // not optional should be defined as a primitive integer.
-    //
-    else if (type == RifColumnType.NUM
-        && columnScale.orElse(Integer.MAX_VALUE) == 0
-        && !isColumnOptional) {
-      return TypeName.INT;
+    } else if (type == RifColumnType.NUM) {
+      if (columnScale.orElse(Integer.MAX_VALUE) > 0) {
+        return ClassName.get(BigDecimal.class);
+      } else if (columnScale.orElse(Integer.MAX_VALUE) == 0) {
+        return isColumnOptional ? ClassName.get(BigDecimal.class) : TypeName.INT;
```

Comment:
Nonblocking question; why does our use of INT or BigDecimal based on if the column is optional?

Reply:
This is an unfortunate holdover that we have to maintain until all data is migrated and the excel spreadsheet changes are totally compllete. In particular it has to do with the original clm_grp_id which was defined as a numeric(12) and which the RifLayoutsProcessor had logic to treat as an integer if the column was 'not null'.. We have to keep it here for a bit as the excel spreadsheet will still have many of the claims table(s) with clm_grp_id defined as numeric(12).

Reply:
Also, as an aside, the only real file that is part of this PR is the V67*.sql; all the other changes are from BFD-1596, that are pulled into this branch. BFD-1596 will be part of the same deploy and it's code changes are what will be in master.